netty/src/main/java/io/grpc/transport/netty/GrpcSslContexts.java

```diff
@@ -93,7 +111,44 @@ public static SslContextBuilder forServer(
    * an application requires particular settings it should override the options set here.
    */
   public static SslContextBuilder configure(SslContextBuilder builder) {
-    return builder.ciphers(Http2SecurityUtil.CIPHERS, SupportedCipherSuiteFilter.INSTANCE)
-        .applicationProtocolConfig(DEFAULT_APN);
+    return configure(builder, SslContext.defaultServerProvider());
+  }
+
+  /**
+   * Set ciphers and APN appropriate for gRPC. Precisely what is set is permitted to change, so if
+   * an application requires particular settings it should override the options set here.
+   */
+  public static SslContextBuilder configure(SslContextBuilder builder, SslProvider provider) {
+    // The default client/server providers in Netty are the same, so it doesn't matter which
+    // method we call.
+    return builder.sslProvider(provider)
```

Comment:
I'm not sure I follow your concern.  Are you uneasy because we're calling a method `SslContext.defaultServerProvider()` for both client and server? I've copied the generic `defaultProvider` code into this class to get rid of that amgibuity.

As far as allowing the application to specify whether they want to use OpenSSL or not, I think we should provide the option regardless of whether or not OpenSSL is installed. They may want to use JDK even though OpenSSL is supported, for example (e.g. maybe they want to enforce ALPN, or maybe there's just an issue they're running into with the OpenSSL implementation).